### PR TITLE
Notification banner long polling and css custom variables

### PR DESCRIPTION
**Pull request by rmathis**
Extends the notification banner to allow more user experience customization such as another icon, css custom variables, and long polling the notifications endpoint for emergency notifications.

### CODE REVIEW

File: @uportal/notification-banner/README.md [hunk 1]
```diff
@@ -27,3 +27,86 @@ npm run build
...
+```html
+<notification-banner notificationIconSize="2x"></notification-icon>
+```
+
+### `debug`
+
+Skips oidc call for api token.
+
+```html
+<notification-banner debug></notification-list
```
Comment (ChristianMurphy) [inline @uportal/notification-banner/README.md:90]:
```suggestion
<notification-banner debug></notification-banner>
```

File: @uportal/notification-banner/README.md [hunk 2]
```diff
@@ -27,3 +27,86 @@ npm run build
...
+```html
+<notification-banner notificationIconSize="2x"></notification-icon>
+```
+
+### `filter`
+
+Query string parameter to append to api call to retrieve notifications.
+
+```html
+<notification-banner notificationIconSize="2x"></notification-icon>
```
Comment (ChristianMurphy) [inline @uportal/notification-banner/README.md:82]:
```suggestion
<notification-banner notificationIconSize="2x"></notification-banner>
```

File: @uportal/notification-banner/README.md [hunk 3]
```diff
@@ -27,3 +27,86 @@ npm run build
...
+```html
+<notification-banner notificationIcon="info"></notification-icon>
+```
+
+### `notificationIconSize`
+
+Icon size for Font Awesome icon displayed next to heading.
+
+```html
+<notification-banner notificationIconSize="2x"></notification-icon>
```
Comment (ChristianMurphy) [inline @uportal/notification-banner/README.md:74]:
```suggestion
<notification-banner notificationIconSize="2x"></notification-banner>
```

File: @uportal/notification-banner/README.md [hunk 4]
```diff
@@ -27,3 +27,86 @@ npm run build
...
+```html
+<notification-banner notificationVariant="info"></notification-icon>
+```
+
+### `notificationIcon`
+
+Icon displayed next to heading. Supports "info" and "exclamation-triangle" from Font Awesome.
+
+```html
+<notification-banner notificationIcon="info"></notification-icon>
```
Comment (ChristianMurphy) [inline @uportal/notification-banner/README.md:66]:
```suggestion
<notification-banner notificationIcon="info"></notification-banner>
```

File: @uportal/notification-banner/README.md [hunk 5]
```diff
@@ -27,3 +27,86 @@ npm run build
...
+```
+
+### Theming
+
+Currently this component supports [CSS Variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_variables) for overriding button colors. Defining the following variables will change the colors for the component accordingly. They will fall back to the colors described below.
+
+You should define this in your custom stylesheet.
+
+fg = foreground (text)
+bg = background
```
Comment (ChristianMurphy) [inline @uportal/notification-banner/README.md:100]: would it make sense to directly use the name over the acronym/abbreviation?

File: @uportal/notification-banner/src/components/NotificationBanner.vue [hunk 6]
```diff
@@ -36,9 +36,25 @@ export default {
...
+            default: 'info'
+        },
+        notificationIcon: {
+            type: String,
+            default: 'info'
+        },
+        notificationIconSize: {
+            type: String,
+            default: '1x'
+        },
```
Comment (ChristianMurphy) [inline @uportal/notification-banner/src/components/NotificationBanner.vue:50]: would it make sense to add validators for these to ensure they are one of the known literals?
https://vuejs.org/v2/guide/components-props.html#Prop-Validation

File: @uportal/notification-banner/README.md [hunk 7]
```diff
@@ -27,3 +27,83 @@ npm run build
...
+
+Currently this component supports [CSS Variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_variables) for overriding button colors. Defining the following variables will change the colors for the component accordingly. They will fall back to the colors described below.
+
+You should define this in your custom stylesheet.
+
+```css
+:root {
+    --notif-banner-heading-text-color: white; /* text color of notification item heading */
+    --notif-banner-body-text-color: white; /* text color of notification item body  */
+    --notif-banner-bg-color: darkred; /* background color of notification items */
```
Comment (ChristianMurphy) [inline @uportal/notification-banner/README.md:103]: would it make sense to spell out background as well?

Comment (rmathis) [reply @uportal/notification-banner/README.md:103]: Personally I think `bg` is pretty well known among developers as shorthand for background and works fine for me. The fg was just me following convention in another component, but honestly didn't make sense to me, so I changed that one.

Comment (ChristianMurphy) [reply @uportal/notification-banner/README.md:103]: makes sense, thanks!

**Pull request merged**